src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.SignedNumber.cs

```diff
@@ -454,5 +610,40 @@ internal void WritePropertyName(long value)
 
             WritePropertyNameUnescaped(utf8PropertyName.Slice(0, bytesWritten));
         }
+
+#if NET7_0_OR_GREATER
```

Comment:
Nit: consider making a partial class with the .NET 7 specific functionality rather than using compile conditionals.

Reply:
I think keeping them grouped as is with extra conditional is fine since extracting them to extra file would be inconsistent with current grouping.

Perhaps current grouping is not ideal and it would be better to split those file into something along the lines of i.e.:
UtfJsonWriter.int.cs
UtfJsonWriter.uint.cs

Possibly we could even generate those files fully or partially but I'll leave this out of scope for this PR since there is a high chance this will get ported to 7.0 and I'd prefer to avoid churning